pkg/proxy/proxy.go

```diff
@@ -411,12 +411,12 @@ func (p *Proxy) ReinstallRoutingRules() error {
 		if err := removeFromEgressProxyRoutesIPv4(); err != nil {
 			return err
 		}
-		if !option.Config.EnableIPSec || option.Config.TunnelingEnabled() {
-			if err := removeFromIngressProxyRoutesIPv4(); err != nil {
+		if (option.Config.EnableEnvoyConfig || option.Config.EnableIPSec) && !option.Config.TunnelingEnabled() {
```

Comment:
nit: Perhaps we could  dry this up into either a convenience call or a var since we're repeating the same conds for ipv4/v6?